dimension/dimension.go

```diff
@@ -1,27 +1,43 @@
 package dimension
 
 import (
+	"context"
 	"encoding/json"
 	"net/http"
 
-	"errors"
 	"io"
 	"io/ioutil"
 
 	errs "github.com/ONSdigital/dp-dataset-api/apierrors"
 	"github.com/ONSdigital/dp-dataset-api/models"
 	"github.com/ONSdigital/dp-dataset-api/store"
+	"github.com/ONSdigital/go-ns/audit"
+	"github.com/ONSdigital/go-ns/common"
 	"github.com/ONSdigital/go-ns/log"
 	"github.com/gorilla/mux"
+	"github.com/pkg/errors"
 )
 
-//Store provides a backend for dimensions
+// Store provides a backend for dimensions
 type Store struct {
+	Auditor audit.AuditorService
 	store.Storer
 }
 
-//GetNodes list from a specified instance
-func (s *Store) GetNodes(w http.ResponseWriter, r *http.Request) {
+// List of audit actions for dimensions
+const (
+	PostDimensionsAction = "postDimensions"
+	PutNodeIDAction      = "putNodeID"
+)
+
+var notFound = map[error]bool{
+	errs.ErrDatasetNotFound:       true,
+	errs.ErrInstanceNotFound:      true,
+	errs.ErrDimensionNodeNotFound: true,
+}
+
+// GetNodesHandler list from a specified instance
+func (s *Store) GetNodesHandler(w http.ResponseWriter, r *http.Request) {
```

Comment:
I see that the dimension endpoints had previously been moved to their own package. Do we intend to move the other dimensions also? If not I would be tempted to move these endpoints back into the API package. I can see that you have had to make a bunch of things package public because of the split, which leaves hardly anything encapsulated within the package. It doesn't feel like its a clean way to split packages. This is not something to change as part of this PR but might be worth a discussion / further task to make things consistent.

Reply:
Im not sure, the dimension package contains instance dimension endpoints, the api package contains version dimension endpoints, hence the weird split.

I would have thought dataset, editions, versions, metadata, observations, dimensions to all be pulled out of the api package in the future into their own packages - and then probably end up renaming the packages (e.g. `dataset` should really be `datasets` for consistency, `dimensions` possibly `versiondimensions` as to not conflict with the current `dimensions` package which then should be renamed to `instancedimensions`)

Or we could go down a different route and nest packages dimensions inside instances and versions, either way you are right we might need a discussion on it.